tests/ui/typeck/ice-with-expr-not-struct-127332.rs

```diff
@@ -0,0 +1,23 @@
+// Regression test for ICE #127332
+
+// Tests that we do not ICE when a with expr
+// is not a struct but something else like an
+// enum in this case
+// This ICE occurred only when the enclosing
+// function was async
+
+
+// Using 2018 edition to supporess an irrelevant
+// error about async fn's
+//@ edition:2018
+
+async fn fun() {
```

Comment:
Why does it need async to reproduce? Can this test be simplified to using a closure? Potentially with the `orig` being declared outside the closure, and captured or even moved into it?

Reply:
You're right. It does not need async. All it needed was a closure like you said. I have updated the test to use a closure now.